### PR TITLE
feat: Typer CLI implementation with cognito device authorization flow

### PR DESCRIPTION
**Pull request by ar-aws**
### Reason for this change

Builders want more flexible methods of interacting with the SBT Control plane and a CLI client provides them an additional way to leverage the platform. 

### Description of changes

- Implemented CLI and published as an PyPI package `sbt-aws-cli`.
- Chose Typer framework due to its enhanced user interface and ease of development. 
- Set up a GitHub Actions workflow to automate the package build and publish process, ensuring that any updates to the CLI are automatically deployed to PyPI upon changes to the `pyproject.toml` file.
- Added documentation in README for the CLI
- Created `cli-auth.ts` which houses resources for OAuth 2.0 device authorization flow to ensure secure authentication for CLI users

### Description of how you validated changes

- Manually tested the CLI commands to ensure they behave as expected and interact correctly with the underlying services.
- Validated the OAuth 2.0 device authorization flow to ensure it handles authentication securely and appropriately.
- Ran the GitHub Actions workflow to confirm that the build and publish process works correctly

- Added bash script to test the new CLI

### Checklist

- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/awslabs/sbt-aws/blob/main/CONTRIBUTING.md)
- [x] I have updated the relevant documentation (if applicable).

---

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.*

### CODE REVIEW

File: docs/public/README.md [hunk 1]
```diff
@@ -635,6 +635,44 @@ The application plane emits this event upon completion of offboarding. Similar t
...
+```bash
+sbt-aws-cli --help
+```
+```bash
+sbt-aws-cli <command> --help
+```
+
+To get started, use the following command to configure the CLI with your Control Plane stack details. The arguments should be directly copied from the stack outputs. Remember you can use the `--help` flag to view the expected arguments.
+```bash
+sbt-aws-cli configure
```
Comment (suhussai) [inline docs/public/README.md:670]: aren't there are args that need to be passed in?

File: scripts/sbt-aws-cli/sbt_aws_cli/main.py [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+    if debug:
+        print(f"Deleting tenant with ID: {tenant_id}")
+
+    response = requests.delete(
+        f"{config['CONTROL_PLANE_API_ENDPOINT']}tenants/{tenant_id}",
+        headers={"Authorization": f"Bearer {config['ACCESS_TOKEN']}"}
+    )
+
+    if debug:
+        print(f"Response: {response.json()}")
```
Comment (suhussai) [inline scripts/sbt-aws-cli/sbt_aws_cli/main.py:284]: we're essentially doing the same thing whether debug mode is enabled or not

File: scripts/sbt-aws-cli/sbt_aws_cli/main.py [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+        "REFRESH_TOKEN": os.getenv('REFRESH_TOKEN')
+    }
+
+    with open(CONFIG_FILE, 'w') as config_file:
+        json.dump(config_data, config_file)
+
+    if debug:
+        print(f"Configuration saved to {CONFIG_FILE}")
+
+def source_config():
```
Comment (suhussai) [inline scripts/sbt-aws-cli/sbt_aws_cli/main.py:141]: i would recommend changing this name. "source" is something that is bash specific, as far as I know.

File: docs/public/README.md (hunk 1 above)
Comment (ar-aws) [reply docs/public/README.md:670]: The idea was to encourage them to use the `--help` flag to better understand the arguments but we could also change that line to this.

`sbt-aws-cli configure <CONTROL_PLANE_STACK> <CLIENT_ID> <FQDN> <CONTROL_PLANE_API_ENDPOINT> <COGNITO_DOMAIN> `

File: scripts/sbt-aws-cli/sbt_aws_cli/main.py (hunk 3 above)
Comment (ar-aws) [reply scripts/sbt-aws-cli/sbt_aws_cli/main.py:141]: How about `initialize_config`

**Comment by suhussai** [comment]
Replaced by PR #93 which is similar but also contains a fix for the failing build job.

**Pull request closed without merging**